client/post-editor/editor-sharing/accordion.jsx

```diff
@@ -46,12 +43,9 @@ const EditorSharingAccordion = React.createClass( {
 		}
 
 		const skipped = PostMetadata.publicizeSkipped( post );
-		const targeted = connections.filter( ( connection ) => {
-			return ! includes( skipped, connection.keyring_connection_ID );
-		} );
-		const targetedServices = serviceConnections.getServicesFromConnections( targeted );
+		const targeted = filter( connections, ( { keyring_connection_ID: ID } ) => ! ( ID in skipped ) );
 
-		return map( targetedServices, 'label' ).join( ', ' );
+		return map( uniqBy( targeted, 'service' ), 'label' ).join( ', ' );
```

Comment:
A possible refactoring to save some iterations (at least if we assume `reduce` + `includes` performs better than `filter` + `uniqBy` + `map`):

```js
return reduce( connections, ( memo, connection ) => {
	const { keyring_connection_ID: id, label } = connection;
	if ( ! includes( skipped, id ) && ! includes( memo, label ) ) {
		memo.push( label );
	}

	return memo;
}, [] ).join( ', ' );
```

(As below, this assumes that we're fine with losing the distinction between `service` and `label`)